src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -514,8 +514,21 @@ HRESULT CProfilerManager::AddInstrumentationMethod(
                 return E_OUTOFMEMORY;
             }
 
+            LoggingFlags flag;
+            pProfilerManagerWrapper->GetInstruMethodLoggingFlags(&flag);
```

Comment:
Does this return an hr?

Reply:
It does return an hr. Seems like you shouldn't try to accumulate logging flags if there is a failure on this call. You could potentially be setting things to garbage.

I don't know if you can assert here, though, since logging may not be initialized yet, and I think asserts log by default.

---
In reply to: [345310744](https://github.com/microsoft/CLRInstrumentationEngine/pull/165#discussion_r345310744) [](ancestors = 345310744)

Reply:
Thanks, I can guard this with an if (SUCCEEDED) check

---
In reply to: [345313778](https://github.com/microsoft/CLRInstrumentationEngine/pull/165#discussion_r345313778) [](ancestors = 345313778,345310744)